src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -160,6 +161,42 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             if (otlpHttpEndpointUrl != null)
             {
                 context.EnvironmentVariables[DashboardConfigNames.DashboardOtlpHttpUrlName.EnvVarName] = otlpHttpEndpointUrl;
+
+                var model = context.ExecutionContext.ServiceProvider.GetRequiredService<DistributedApplicationModel>();
+                var allResourceEndpoints = model.Resources
+                    .Where(r => !string.Equals(r.Name, KnownResourceNames.AspireDashboard, StringComparisons.ResourceName))
+                    .SelectMany(r => r.Annotations)
+                    .OfType<EndpointAnnotation>()
+                    .ToList();
+
+                var corsOrigins = new HashSet<string>();
+                foreach (var endpoint in allResourceEndpoints)
+                {
+                    if (endpoint.UriScheme is "http" or "https")
+                    {
+                        // Prefer allocated endpoint over EndpointAnnotation.Port.
+                        var origin = endpoint.AllocatedEndpoint?.UriString;
+                        if (origin == null && endpoint.Port != null)
```

Comment:
Not sure if this fallback logic is necessary. We should always have an allocated endpoint at this stage.